src/Spatial/Euclidean/Circle2D.cs

```diff
@@ -131,6 +134,63 @@ public static Circle2D FromPoints(Point2D pointA, Point2D pointB, Point2D pointC
             return new Circle2D(center, radius);            
         }
 
+        /// <summary>
+        /// Returns intersection a point2D array if this circle and the given line have the intersections
+        /// </summary>
+        /// <param name="line">the given line</param>
+        /// <returns>intersections as a Point2D Array, depending on the count.</returns>
+        public Point2D[] IntersectWith(Line2D line)
+        {
+            var ts = this.findParameterTs(line);
+            var result = ts
+                .Select(t => line.StartPoint + t * line.Direction)
+                .ToArray();
+            return result;
+        }
+
+        private double[] findParameterTs(Line2D line)
+        {
+            // These 2 equations in vector form can be described
+            // (p-cc)^2=r^2 (eq1)
+            // p=s+t*d     (eq2)
+            // , where p is the point on the line and/or circle,
```

Comment:
_where p is the point on the line, d is the line direction_

Reply:
done in the commit https://github.com/f-frhs/mathnet-spatial/commit/c2612b61fa2c6f705522c5926f379ec404a77ca4

FYI, there is `c` as it was.
Replacing `c` to `cc` is another problem.